.github/workflows/regression-tests.yml

```diff
@@ -32,6 +32,7 @@ jobs:
       run: |
         cd regression-tests
         bash run-tests.sh -c ${{ matrix.compiler }}
+      continue-on-error: true
```

Comment:
> Prevents a workflow run from failing when a job fails. Set to true to allow a workflow run to pass when this job fails.

This will make the run pass.  I think you need to make this upload a new job and have it run always.  https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds

Reply:
Another issue unrelated to this PR, the job is called `regression-tests-linux-mac` but also runs Windows.

Reply:
My bad, I must have forgotten to update.
You can fix it in is this PR.

Reply:
Can we really use [jobs.<job_id>.needs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds)?
We need to access the output patch file on the job.

Reply:
Yeah, I guess I don't know.  I just know that the `continue-on-error` means ignore failures, so that won't do what we want.

Reply:
Maybe the diff needs to be a job output, not sure how that works for matrix jobs.  https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idoutputs

Reply:
I think there is a simpler solution. I am testing it now.

Reply:
There is one solution in #935